src/dmd/root/filename.d

```diff
@@ -1002,26 +1002,15 @@ nothrow:
                 // First find out how long the buffer has to be.
                 const fullPathLength = GetFullPathNameW(&wname[0], 0, null, null);
                 if (!fullPathLength) return null;
-                auto fullPath = new wchar[fullPathLength];
+                auto fullPath = (cast(wchar*) mem.xmalloc_noscan((fullPathLength + 1) * wchar.sizeof))[0 .. fullPathLength + 1];
+                scope(exit) mem.xfree(fullPath.ptr);
 
                 // Actually get the full path name
-                const fullPathLengthNoTerminator = GetFullPathNameW(
-                    &wname[0], fullPathLength, &fullPath[0], null /*filePart*/);
-                // Unfortunately, when the buffer is large enough the return value is the number of characters
-                // _not_ counting the null terminator, so fullPathLengthNoTerminator should be smaller
-                assert(fullPathLength > fullPathLengthNoTerminator);
-
-                // Find out size of the converted string
-                const retLength = WideCharToMultiByte(
-                    codepage, 0 /*flags*/, &fullPath[0], fullPathLength, null, 0, null, null);
-                auto ret = new char[retLength];
```

Comment:
This previously meant that the result could **not** be free'd via `FileName.free()`.